-e  in requirements.txt not handled properly

```
>>> import requirements
>>> for req in requirements.parse("-e ."): print(req)
... 

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/requirements/parser.py", line 87, in parse
    yield Requirement.parse(line)
  File "/usr/local/lib/python3.9/site-packages/requirements/requirement.py", line 251, in parse
    return cls.parse_editable(
  File "/usr/local/lib/python3.9/site-packages/requirements/requirement.py", line 157, in parse_editable
    assert local_match is not None, 'This should match everything'
AssertionError: This should match everything

```

Also

```
>> for req in requirements.parse("-e abc"): print(req)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/requirements/parser.py", line 87, in parse
    yield Requirement.parse(line)
  File "/usr/local/lib/python3.9/site-packages/requirements/requirement.py", line 251, in parse
    return cls.parse_editable(
  File "/usr/local/lib/python3.9/site-packages/requirements/requirement.py", line 157, in parse_editable
    assert local_match is not None, 'This should match everything'
AssertionError: This should match everything
```

The first example can be found in https://github.com/jerryjliu/llama_index and pip seems to allow https://pip.pypa.io/en/stable/reference/requirements-file-format/

--- a/requirements/requirement.py
+++ b/requirements/requirement.py
@@ -148,13 +148,12 @@ class Requirement:
                 fragment = parse_fragment(groups['fragment'])
                 egg = cast(str, fragment.get('egg'))
                 req.name, req.extras = parse_extras_require(egg)  # type: ignore
-                req.hash_name, req.hash = get_hash_info(fragment)     # type: ignore
+                req.hash_name, req.hash = get_hash_info(fragment)  # type: ignore
                 req.subdirectory = fragment.get('subdirectory')  # type: ignore
             for vcs in VCS:
                 if str(req.uri).startswith(vcs):
                     req.vcs = vcs  # type: ignore
-        else:
-            assert local_match is not None, 'This should match everything'
+        elif local_match is not None:
             groups = local_match.groupdict()
             req.local_file = True
             if groups['fragment']:
@@ -164,6 +163,9 @@ class Requirement:
                 req.hash_name, req.hash = get_hash_info(fragment)  # type: ignore
                 req.subdirectory = fragment.get('subdirectory')  # type: ignore
             req.path = cast(str, groups['path'])  # type: ignore
+        else:
+            req.local_file = True
+            req.path, req.name = line.rsplit('/', 1)  # type: ignore
 
         return req
 

--- a/tests/reqfiles/editable_1.expected
+++ b/tests/reqfiles/editable_1.expected
@@ -62,5 +62,21 @@
   "hash": null,
   "specs": [],
   "revision": null
- }
+ },
+ {
+  "specifier": false,
+  "local_file": true,
+  "name": "package_name",
+  "editable": true,
+  "subdirectory": null,
+  "uri": null,
+  "extras": [],
+  "vcs": null,
+  "path": "path/to/local",
+  "line":  "-e path/to/local/package_name",
+  "hash_name": null,
+  "hash": null,
+  "specs": [],
+  "revision": null
+  }
 ]

--- a/tests/reqfiles/editable_1.txt
+++ b/tests/reqfiles/editable_1.txt
@@ -2,3 +2,4 @@
 --editable git://git.myproject.org/MyProject.git@da39a3ee5e6b4b0d3255bfef95601890afd80709#egg=MyProject
 --editable hg+http://hg.myproject.org/MyProject/@special_feature#egg=MyProject
 -e bzr+lp://MyProject#egg=MyProject
+-e path/to/local/package_name

